codec/decode.go

```diff
@@ -544,10 +549,12 @@ func (sd *Decoder) DecodeTuple(t interface{}) (interface{}, error) {
 				ptr := fieldValue.(*[]string)
 				*ptr = o.([]string)
 			default:
-				_, err = sd.Decode(v.Field(i).Interface())
+				o, err = sd.Decode(v.Field(i).Interface())
 				if err != nil {
 					break
 				}
+				// TODO: clean up this function, can use field.Set everywhere (remove switch case?)
```

Comment:
Did you mean to address in this pull request or is there an issue open for this?

Reply:
yes #611